soc/arc/snps_nsim/CMakeLists.txt

```
@@ -23,6 +23,16 @@ else()
 			       -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
 			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
 
+  zephyr_compile_options_ifdef(CONFIG_SOC_NSIM_EM11D -arcv2em -core3 -Xdiv_rem=radix2
+			       -Xbitscan -Xswap -Xbarrel_shifter
+			       -Xshift_assist -Xfpus_div -Xfpu_mac -Xfpuda -Xfpus_mpy_slow
+			       -Xfpus_div_slow -Xbitstream -Xtimer0 -Xtimer1)
+
+  if(CONFIG_SOC_NSIM_EM11D)
+    set(ENV{Z_ARC_DSP_OPTIONS} "-Xxy;-Xagu_large;-Hfxapi;-Xdsp2;-Xdsp_complex;-Xdsp_accshift=full;\
+-Xdsp_divsqrt=radix2;-Xdsp_itu;-Xdsp_ctrl=postshift,noguard,convergent;-Xmpy_option=mpyd")
```

I'd rather avoid environment usage for internal variables. We can use properties instead. I.e:

```suggestion
    set_property(GLOBAL PROPERTY z_arc_dsp_options "-Xxy;-Xagu_large;-Hfxapi;-Xdsp2;-Xdsp_complex;-Xdsp_accshift=full;\
-Xdsp_divsqrt=radix2;-Xdsp_itu;-Xdsp_ctrl=postshift,noguard,convergent;-Xmpy_option=mpyd")
```